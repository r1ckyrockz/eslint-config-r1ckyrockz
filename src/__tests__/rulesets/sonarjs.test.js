const { createSonarjsRules } = require('../../rulesets/sonarjs');

describe('createSonarjsRules', () => {
  test('matches snapshot without typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createSonarjsRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createSonarjsRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'sonarjs/cognitive-complexity';
    const level = 'off';

    const project = {
      rules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    const defaultRuleset = createSonarjsRules({
      ...project,
      rules: {},
    });

    const result = createSonarjsRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
