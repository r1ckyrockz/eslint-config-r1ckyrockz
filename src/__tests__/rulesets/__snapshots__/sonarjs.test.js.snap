// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSonarjsRules allows passing custom rules 1`] = `
Object {
  "sonarjs/cognitive-complexity": "off",
  "sonarjs/max-switch-cases": Array [
    "error",
    15,
  ],
  "sonarjs/no-all-duplicated-branches": "error",
  "sonarjs/no-collapsible-if": "error",
  "sonarjs/no-collection-size-mischeck": "warn",
  "sonarjs/no-duplicate-string": "error",
  "sonarjs/no-duplicated-branches": "error",
  "sonarjs/no-element-overwrite": "error",
  "sonarjs/no-extra-arguments": "off",
  "sonarjs/no-identical-conditions": "error",
  "sonarjs/no-identical-expressions": "error",
  "sonarjs/no-identical-functions": "error",
  "sonarjs/no-inverted-boolean-check": "error",
  "sonarjs/no-one-iteration-loop": "error",
  "sonarjs/no-redundant-boolean": "error",
  "sonarjs/no-redundant-jump": "error",
  "sonarjs/no-same-line-conditional": "off",
  "sonarjs/no-small-switch": "warn",
  "sonarjs/no-unused-collection": "error",
  "sonarjs/no-use-of-empty-return-value": "error",
  "sonarjs/no-useless-catch": "error",
  "sonarjs/prefer-immediate-return": "warn",
  "sonarjs/prefer-object-literal": "error",
  "sonarjs/prefer-single-boolean-return": "error",
  "sonarjs/prefer-while": "error",
}
`;

exports[`createSonarjsRules matches snapshot with typescript 1`] = `
Object {
  "sonarjs/cognitive-complexity": "warn",
  "sonarjs/max-switch-cases": Array [
    "error",
    15,
  ],
  "sonarjs/no-all-duplicated-branches": "error",
  "sonarjs/no-collapsible-if": "error",
  "sonarjs/no-collection-size-mischeck": "warn",
  "sonarjs/no-duplicate-string": "error",
  "sonarjs/no-duplicated-branches": "error",
  "sonarjs/no-element-overwrite": "error",
  "sonarjs/no-extra-arguments": "off",
  "sonarjs/no-identical-conditions": "error",
  "sonarjs/no-identical-expressions": "error",
  "sonarjs/no-identical-functions": "error",
  "sonarjs/no-inverted-boolean-check": "error",
  "sonarjs/no-one-iteration-loop": "error",
  "sonarjs/no-redundant-boolean": "error",
  "sonarjs/no-redundant-jump": "error",
  "sonarjs/no-same-line-conditional": "off",
  "sonarjs/no-small-switch": "warn",
  "sonarjs/no-unused-collection": "error",
  "sonarjs/no-use-of-empty-return-value": "error",
  "sonarjs/no-useless-catch": "error",
  "sonarjs/prefer-immediate-return": "warn",
  "sonarjs/prefer-object-literal": "error",
  "sonarjs/prefer-single-boolean-return": "error",
  "sonarjs/prefer-while": "error",
}
`;

exports[`createSonarjsRules matches snapshot without typescript 1`] = `
Object {
  "sonarjs/cognitive-complexity": "warn",
  "sonarjs/max-switch-cases": Array [
    "error",
    15,
  ],
  "sonarjs/no-all-duplicated-branches": "error",
  "sonarjs/no-collapsible-if": "error",
  "sonarjs/no-collection-size-mischeck": "warn",
  "sonarjs/no-duplicate-string": "error",
  "sonarjs/no-duplicated-branches": "error",
  "sonarjs/no-element-overwrite": "error",
  "sonarjs/no-extra-arguments": "error",
  "sonarjs/no-identical-conditions": "error",
  "sonarjs/no-identical-expressions": "error",
  "sonarjs/no-identical-functions": "error",
  "sonarjs/no-inverted-boolean-check": "error",
  "sonarjs/no-one-iteration-loop": "error",
  "sonarjs/no-redundant-boolean": "error",
  "sonarjs/no-redundant-jump": "error",
  "sonarjs/no-same-line-conditional": "off",
  "sonarjs/no-small-switch": "warn",
  "sonarjs/no-unused-collection": "error",
  "sonarjs/no-use-of-empty-return-value": "error",
  "sonarjs/no-useless-catch": "error",
  "sonarjs/prefer-immediate-return": "warn",
  "sonarjs/prefer-object-literal": "error",
  "sonarjs/prefer-single-boolean-return": "error",
  "sonarjs/prefer-while": "error",
}
`;
